File: recipe-java-runtime/src/main/java/ca/derekcormier/recipe/Payload.java
package ca.derekcormier.recipe;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;

public class Payload {
    private Recipe recipe;
    private Cake cake;

    @JsonCreator
    public Payload(
        @JsonProperty("recipe") Recipe recipe,
        @JsonProperty("cake") Cake cake
    ) {
        this.recipe = recipe;
        this.cake = cake == null ? new Cake(): cake;
    }
    public Ingredient getRecipe() {
        return recipe;
    }

    public Cake getCake() {
        return cake;
    }
}

File: recipe-ts-runtime/src/Payload.ts
import { Recipe } from "./Recipe";
import { Cake } from "./Cake";

export class Payload {
    constructor(private recipe: Recipe, private cake: Cake) {}

    public static fromJSON(json: any, ingredientTypes: {[key: string]: Function}): Payload {
        return new Payload(Recipe.fromJSON(json.recipe, ingredientTypes), Cake.fromJSON(json.cake));
    }

    public getRecipe(): Recipe {
        return this.recipe;
    }

    public getCake(): Cake {
        return this.cake;
    }
}